nextflow

```diff
@@ -394,8 +394,13 @@ else
     cli=($("$JAVA_CMD" "${JAVA_OPTS[@]}" -jar "$NXF_BIN"))
     [[ $? -ne 0 ]] && echo_red 'Unable to initialize nextflow environment' && exit 1
 
+    pat='"([^"]*)"(.*)'
+    [[ "${cli[@]}" =~ $pat ]]
+    cmd=(${BASH_REMATCH[1]})
+    rest=(${BASH_REMATCH[2]})
+
     if [[ "$JAVA_VER" =~ ^(9|10|11|12|13|14|15|16|17) ]]; then
```

Comment:
We have a new Bash guru! 😄

Reply:
Kidding apart what's the logic of the snippet above?

Reply:
yes, sorry for the lack of explanations

main problem comes because `cli=($("$JAVA_CMD" "${JAVA_OPTS[@]}" -jar "$NXF_BIN"))` split all the command in an array but bash uses the blank as separator so `launcher=("${cli[@]:0:1}")` assign an incomplete path 

the idea is to apply a regexp and extract the element between double quotes into `cmd` var and the rest into the `rest`var and after we can create the array as doing previously

Reply:
That's smart, let's add this explanation in the script as a comment. thanks!

Reply:
done, let me know if it's enough or need more explanation